exonum/benches/criterion/indexes.rs

```diff
@@ -0,0 +1,158 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+static KEY_SET_FAMILY_NAME: &str = "key";
+static VALUE_SET_FAMILY_NAME: &str = "key";
+
+use criterion::{
+    AxisScale, Bencher, Criterion, ParameterizedBenchmark, PlotConfiguration, Throughput,
+};
+use exonum::{crypto, storage::{KeySetIndex, ValueSetIndex, Database, DbOptions, RocksDB}};
+use tempdir::TempDir;
+use num::pow::pow;
+
+use std::{fs, path::{Path, PathBuf}};
+
+pub fn key_value_set_index(c: &mut Criterion) {
+    crypto::init();
+
+    let tempdir = TempDir::new("exonum").unwrap();
+
+    let path = tempdir.path().to_path_buf();
+    c.bench(
+        "insert",
+        ParameterizedBenchmark::new("key_set_index", move |bencher, key_size| {
+            key_set_insert(path.clone(), bencher, key_size)
+        }, (0..16).map(|i| pow(2, i)))
+            .throughput(|s| Throughput::Bytes(*s as u32))
+            .plot_config(PlotConfiguration::default().summary_scale(AxisScale::Logarithmic)),
+    );
+    fs::remove_dir_all(tempdir.path()).unwrap();
+    let path = tempdir.path().to_path_buf();
+    c.bench(
+        "insert",
+        ParameterizedBenchmark::new("value_set_index", move |bencher, key_size| {
+            value_set_insert(path.clone(), bencher, key_size)
+        }, (6..16).map(|i| pow(2, i)))
+            .throughput(|s| Throughput::Bytes(*s as u32))
+            .plot_config(PlotConfiguration::default().summary_scale(AxisScale::Logarithmic)),
+    );
+    fs::remove_dir_all(tempdir.path()).unwrap();
+
+    let path = tempdir.path().to_path_buf();
+    c.bench(
+        "contains",
+        ParameterizedBenchmark::new("key_set_index", move |bencher, key_size| {
+            key_set_contains(path.clone(), bencher, key_size)
+        }, (6..16).map(|i| pow(2, i)))
+            .throughput(|s| Throughput::Bytes(*s as u32))
+            .plot_config(PlotConfiguration::default().summary_scale(AxisScale::Logarithmic)),
+    );
+    fs::remove_dir_all(tempdir.path()).unwrap();
+    let path = tempdir.path().to_path_buf();
+    c.bench(
+        "contains",
+        ParameterizedBenchmark::new("value_set_index", move |bencher, key_size| {
+            value_set_contains(path.clone(), bencher, key_size)
+        }, (6..16).map(|i| pow(2, i)))
+            .throughput(|s| Throughput::Bytes(*s as u32))
+            .plot_config(PlotConfiguration::default().summary_scale(AxisScale::Logarithmic)),
+    );
+    fs::remove_dir_all(tempdir.path()).unwrap();
+    let path = tempdir.path().to_path_buf();
+    c.bench(
+        "contains",
+        ParameterizedBenchmark::new("value_set_index: contains_by_hash", move |bencher, key_size| {
+            value_set_contains_by_hash(path.clone(), bencher, key_size)
+        }, (6..16).map(|i| pow(2, i)))
```

Comment:
we can move this iterator somewhere, it's looking like reusable

Reply:
Probably benchmark initialisation can be mooved too

```rust
fn create_benchmark<F>(group:&str, name:&str, F)
where F:FnMut(&Path, &mut Bencher, &usize) 
```

Or you didn't found it usefull?